### DISCUSSION TITLE
Is it possible to run pymodbus simulation with multipy slaves?

### DISCUSSION THREAD

**Discussion by 0x975** [pymodbus-dev/pymodbus#1938] [Help] [answered]
I want to run pymodbus simulation (with http ui) with multiple slaves from JSON configuration.
is it tecnicaly possible?

**Comment by janiversen** [comment, marked as answer]
Yes read the documentation.

**Reply by yurybura** [reply]
Thank you for the answer.

To be honest, I read the section https://pymodbus.readthedocs.io/en/latest/source/library/simulator/config.html, but didn't see how to define a slave address. 

The documentation contains very brief description like:
> You can define as many server and devices as you like, when starting [pymodbus.simulator](https://pymodbus.readthedocs.io/en/latest/source/library/simulator/web.html#pymodbus-simulator) you select one server and one device to simulate.

What kind of documentation section contains the description how to start simulation for multiple slaves connected to the one serial port?

**Reply by janiversen** [reply]
One server simulates 1 device, not multiple.

Pull requests are welcome if you want to change that.

**Reply by yurybura** [reply]
That means that the simulation of multiple devices is not possible. It is clear.
How to define slave address in Simulator JSON?

**Reply by janiversen** [reply]
It is possible to simulate multiple devices, just define multiple servers. You do not define the slave address because it is currently not needed.

**Reply by yurybura** [reply]
Every Modbus RTU telegram contains slave address. For RS485 connection different slave addresses are mandatory.

**Reply by yurybura** [reply]
> It is possible to simulate multiple devices, just define multiple servers.

It isn't possible to open serial port several times. RS485 connection means one Modbus Master on the one side and multiple Modbus Slaves on the second side.

**Reply by janiversen** [reply]
The simulator simulates 1 device for each server, that is documented !! The simulator do NOT simulate n devices for each server, if you want that kind of functionality pull requests are welcome.

**Comment by janiversen** [comment]
You can easily define n servers with different com ports, and then connect each com port to the rs485 cable.

Just a head up, a server with only 1 com port, cannot simulate n slaves on the same rs485 line...simply because it is only have one connection, so a lot of tests would be wrong.

in tcp a server be have n slaves connected, but the server act as 1 devices with n devices connected below it.
